crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -106,6 +132,8 @@ impl<'interner> TypeChecker<'interner> {
             }
             HirExpression::Index(index_expr) => self.check_index_expression(index_expr),
             HirExpression::Call(call_expr) => {
+                self.maybe_deprecated(&call_expr.func);
```

Comment:
We should rename this to something like `check_if_deprecated`. Also, doing this in the typechecker seems a bit odd. Can we do it during name resolution?

One more thing - part of the reason why the check in `maybe_deprecated` is more complex than it needs to be is that we're checking at the function callsite rather than where the function variable is referenced in the `Ident` case. So we should probably move the check to when an identifier is resolved to a function during name resolution.

Reply:
I first tried at the name resolution step, but the code was panicking due to the missing function signature.